webrtc.html

```diff
@@ -2582,6 +2636,110 @@
+      </p>
+      <p>
+        The explicit certificate management functions provided here are
+        optional.  If an application does not provide
+        the <code>certificates</code> configuration option when constructing
+        an <code>RTCPeerConnection</code> a new set of certificates MUST be
+        generated by the <a>user agent</a>.  That set MUST include an ECDSA
+        certificate with a private key on the P-256 curve and a signature with a
+        SHA-256 hash.
+      </p>
+
+      <dl class="idl" title="partial interface RTCPeerConnection">
+        <dt>static Promise&lt;RTCCertificate&gt; generateCertificate
+          (AlgorithmIdentifier keygenAlgorithm)</dt>
+        <dd>
+          <p>This function takes an argument that describes how the private key associated with the certificate is generated.  The <var>keygenAlgorithm</var> argument uses the WebCrypto
+          [[!WebCryptoAPI]] <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#dfn-AlgorithmIdentifier">AlgorithmIdentifier</a>
+          type.  The <var>keygenAlgorithm</var> value MUST be a valid argument
+            to <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#SubtleCrypto-method-generateKey"><code>Crypto.subtle.generateKey</code></a>.</p>
+
+          <p>The selected <code>AlgorithmIdentifier</code> MUST be valid for
+          both key generation and signing.  The following values are valid:
+            <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#rsassa-pkcs1">RSASSA-PKCS1-v1_5</a>,
+            <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#rsa-pss">RSA-PSS</a>,
+            <!-- (note, PSS is not useful with current TLS but is included for future-proofing), -->
+            and <a href="https://dvcs.w3.org/hg/webcrypto-api/raw-file/tip/spec/Overview.html#ecdsa">ECDSA</a>.</p>
+
+          <p>The resulting certificate contains a signature.  The validity of
+          this signature is only relevant for compatibility reasons.  Only the
+          public key and the resulting certificate fingerprint are used for
+          WebRTC, but it is more likely that the certificate will be accepted if
+          the certificate is well formed.  The browser selects the algorithm
+          used to sign the certificate; a browser SHOULD select SHA-256 if a
+          hash algorithm is needed.</p>
+
+          <p>The resulting certificate MUST NOT include information that can be
+          linked to a user or <a>user agent</a>.  Randomized values for
+          distinguished name and serial number SHOULD be used.</p>
+
+          <p>A <a>user agent</a> MUST reject a call
+          to <code>generateCertificate()</code> that uses an algorithm that is
+          not supported: either because the algorithms are not supported or it
+          cannot use a certificate produced by those algorithms
+          with <code>RTCPeerConnection</code>.</p>
```

Comment:
Do you think that it is impossible to work out how to thread that needle?  It might mean knowing what you support in the various involved components, but that's hardly special.  All this takes is identifying what options work and rejecting all others.

Why do you think we have to identify what parts of the browser internals need to support anything?  The browser either supports X or it doesn't.  (Again, WebCrypto has no play here, the spec is only being used because it provides a reusable method for identifying key generation algorithms and parameters.)